### PR TITLE
#583_User_Email_Search

### PR DESCRIPTION
**Pull request by lnzd17**
PR for ticket #583 
* name and email can be searched from the same input field
* created controller test

One thing to note: When searching by email a full email address has to be submitted. It does not work with a 'partial' address. Did we want to have that work? 

I was trying to figure out how to 'clean' up the `params` if statement inside the `index` method. Would love to hear suggestions on how to make it a bit tidier if anything comes to mind.

### CODE REVIEW

**Comment by markottaviani** [comment]
Great work thanks!   @TheDudeWithTheThing any ideas on how you might tidy up the users_controller?  I'm not sure either 😄

**Comment by TheDudeWithTheThing** [comment]
come on now, all that stuff in the controller should be moved to a UserSearchService object, the end result would look glorious

```
@users = UserSearchService.search(params)
```

**Comment by TheDudeWithTheThing** [comment]
@lnzd17 there's already a DogSearch class you can "mimic"

**Comment by markottaviani** [comment]
haha, of course.  @lnzd17 I'm sure that'd be a good experience for you if you feel like taking a shot at it.  

We also have #192 looming somewhere in the future, which would make a lot of work here moot.

**Comment by lnzd17** [comment]
I'll see if I can `hack it`. :)

**Comment by lnzd17** [comment]
Created UserSearch class. I will work on tests now. I think this is on the right track. Holler if not :)

File: app/models/user_searcher.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class UserSearcher
+  PER_PAGE = 30
+  EMAIL_CHECK = /\A([\w+\-].?)+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+
+  def initialize(params: {})
+    @params = params
+  end
+
+  def search
+    @users = User.all
```
Comment (TheDudeWithTheThing) [inline app/models/user_searcher.rb]: no need for the `all` part here

Comment (lnzd17) [inline app/models/user_searcher.rb]: 👍

**Comment by TheDudeWithTheThing** [comment]
this looks great so far, nice work!

**Comment by lnzd17** [comment]
ready for review

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <tr>
   <td><%= link_to user.name, user %></td>
+  <td><%= user.email %></td>
```
Comment (markottaviani) [inline app/views/users/_user.html.erb]: Given that an email search is only going to return one User, and the limited real-estate on the table view,  I'm thinking that we don't display the email address in the table view.

Comment (lnzd17) [inline app/views/users/_user.html.erb]: That makes sense. I can revise.

**Pull request merged**